test/run_tests.py

```diff
@@ -96,68 +112,110 @@ def http_server():
-
-
-def get_travis_conf(browser):
-    conf = [
-        '--variable', 'SAUCE_USERNAME:{}'.format(env.SAUCE_USERNAME),
-        '--variable', 'SAUCE_ACCESS_KEY:{}'.format(env.SAUCE_ACCESS_KEY)
-    ]
-    if browser == 'firefox':
-        conf.extend([
-            '--variable',
-            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
-                env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY
-            )
-        ])
-    else:
-        conf.extend([
-            '--variable',
-            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
-                env.TRAVIS_JOB_NUMBER, browser
-            )
-        ])
-    return conf
+    subprocess.call(
+        command, env=dict(os.environ, ROBOT_SYSLOG_FILE=syslog)
+    )
 
 
-def log_start(command_list):
+def log_start(command_list, sauceusername, saucekey):
     command = subprocess.list2cmdline(command_list)
-    for hidden in [env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY]:
+    for hidden in [sauceusername, saucekey]:
         if hidden:
             command = command.replace(hidden, '*' * len(hidden))
     print()
     print('Starting test execution with command:')
     print(command)
 
 
-def process_output(browser, cli_options):
+def get_sauce_conf(browser, sauceusername, saucekey):
+    if browser == 'chrome' and env.TRAVIS:
+        conf = []
+    else:
+        conf = [
+            '--variable', 'SAUCE_USERNAME:{}'.format(sauceusername),
+            '--variable', 'SAUCE_ACCESS_KEY:{}'.format(saucekey),
+            '--variable',
+            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
+                sauceusername, saucekey
+            ),
+            '--variable',
+            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
+                env.TRAVIS_JOB_NUMBER, browser
+            )
+        ]
+    return conf
+
+
+def process_output(browser, suite):
     print('Verifying results...')
     output = os.path.join(env.RESULTS_DIR, 'output.xml')
     robotstatuschecker.process_output(output, verbose=False)
     options = [opt.format(browser=browser) for opt in REBOT_OPTIONS]
+    if suite:
+        options += ['-s', suite]
     try:
-        rebot_cli(options + cli_options + [output])
+        rebot_cli(options + [output])
     except SystemExit as exit:
         return exit.code
 
 
+def sauce_credentials(sauceusername, saucekey):
+    if env.TRAVIS:
+        username = env.SAUCE_USERNAME
+        key = env.SAUCE_ACCESS_KEY
+    else:
+        username = sauceusername
+        key = saucekey
+    return username, key
+
+
 if __name__ == '__main__':
-    if len(sys.argv) < 3 or '--help' in sys.argv:
-        sys.exit(__doc__)
+    parser = argparse.ArgumentParser(description='Library test runner')
+    parser.add_argument(
+        '--interpreter',
+        default='python',
+        help='Interpreter used run the test'
+    )
+    parser.add_argument('browser', help='Browser used in testing')
```

Comment:
I kind a liked to this way. For me it makes the travis [matrix](https://github.com/robotframework/Selenium2Library/pull/716/files#diff-354f30a63fb0907d4ad57269548329e3R20) easier to read. Also it saves few letter for typing, now it can be used with: 
`run_tests.py chrome`
instead of:
`run_tests.py -B chrome`